v2/kafka-to-pubsub/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToPubsub.java

```diff
@@ -0,0 +1,177 @@
+ *       --image-gcr-path "${TARGET_GCR_IMAGE}" \
+ *       --sdk-language "JAVA" \
+ *       --flex-template-base-image ${BASE_CONTAINER_IMAGE} \
+ *       --metadata-file "src/main/resources/kafka_to_pubsub_metadata.json" \
+ *       --jar "cd /path/to/DataflowTemplates/v2/kafka-to-pubsub" \
+ *       --env FLEX_TEMPLATE_JAVA_MAIN_CLASS="org.apache.beam.templates.KafkaToPubsub"
+ *
+ * # Execute template:
+ * API_ROOT_URL="https://dataflow.googleapis.com"
+ * TEMPLATES_LAUNCH_API="${API_ROOT_URL}/v1b3/projects/${PROJECT}/templates:launch"
+ * JOB_NAME="kafka-to-pubsub-`date +%Y%m%d-%H%M%S-%N`"
+ *
+ * time curl -X POST -H "Content-Type: application/json"     \
+ *     -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+ *     "${TEMPLATES_LAUNCH_API}"`
+ *     `"?validateOnly=false"`
+ *     `"&dynamicTemplate.gcsPath=${BUCKET_NAME}/path/to/image-spec"`
+ *     `"&dynamicTemplate.stagingLocation=${BUCKET_NAME}/staging" \
+ *     -d '
+ *      {
+ *       "jobName":"${JOB_NAME}",
+ *       "parameters": {
+ *           "bootstrapServers":"broker_1:9092,broker_2:9092",
+ *           "inputTopics":"topic1,topic2",
+ *           "outputTopic":"projects/${PROJECT}/topics/your-topic-name"
+ *        }
+ *       }
+ *      '
+ * </pre>
+ */
+public class KafkaToPubsub {
+
+  /* Logger for class.*/
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaToPubsub.class);
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    KafkaToPubsubOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(KafkaToPubsubOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs a pipeline which reads message from Kafka and writes it to GCS.
+   *
+   * @param options arguments to the pipeline
+   */
+  public static PipelineResult run(KafkaToPubsubOptions options) {
+
+    List<String> topicsList = new ArrayList<>(Arrays.asList(options.getInputTopics().split(",")));
+
+    checkArgument(
+        topicsList.size() > 0 && topicsList.get(0).length() > 0,
+        "inputTopics cannot be an empty string.");
+
+    List<String> bootstrapServersList =
+        new ArrayList<>(Arrays.asList(options.getBootstrapServers().split(",")));
+
+    checkArgument(
+        bootstrapServersList.size() > 0 && topicsList.get(0).length() > 0,
+        "bootstrapServers cannot be an empty string.");
+
+    // Create the pipeline
+    Pipeline pipeline = Pipeline.create(options);
+    LOG.info(
+        "Starting Kafka-To-PubSub pipeline with parameters bootstrap servers:"
+            + options.getBootstrapServers()
+            + " input topics: "
+            + options.getInputTopics()
+            + " output pubsub topic: "
+            + options.getOutputTopic());
+
+    /*
+     * Steps:
+     *  1) Read messages in from Kafka
+     *  3) Write successful records to PubSub
```

Comment:
Steps:
     *  1) Read messages in key value format from Kafka
     *  2) Extract Values Only
     *  3) Write successful records to PubSub

Reply:
I added udf to the pipeline. Now it looks like:
```java
/*
         * Steps:
         *  1) Read messages in from Kafka
         *  2) Extract Values Only
         *  3) Transform message payload via UDF
         *  4) Write successful records out to Pub/Sub
         *  5) Write failed records out to Pub/Sub failure topic
         */
```
is it well?